client/question/services/ConversationManagerServiceTestUtils.js

```diff
@@ -79,38 +70,30 @@ tie.factory('ExpectedFeedbackObjectFactory', [function() {
    */
   ExpectedFeedback.prototype.verifyFeedback = function(submissionResult) {
     var errorMessages = [];
+    var observedFeedback = submissionResult.getFeedback();
+    var observedFeedbackParagraphObject = observedFeedback.getParagraphs();
+    var observedStdout = submissionResult.getStdout();
+    var expectedStdout = this._stdout;
 
-    var feedback = submissionResult.getFeedback();
-
-    if (this._firstFeedbackParagraphContent !== null) {
-      var observedContent = feedback.getParagraphs()[0].getContent();
-      var expectedContent = this._firstFeedbackParagraphContent;
-      if (observedContent !== expectedContent) {
-        errorMessages.push(
-          'Bad content for first feedback paragraph: expected "' +
-          expectedContent + '" but received "' + observedContent + '"');
-      }
-    }
-
-    if (this._secondFeedbackParagraphContent !== null) {
-      observedContent = feedback.getParagraphs()[1].getContent();
-      expectedContent = this._secondFeedbackParagraphContent;
-      if (observedContent !== expectedContent) {
-        errorMessages.push(
-          'Bad content for second feedback paragraph: expected "' +
-          expectedContent + '" but received "' + observedContent + '"');
+    for (var i in this._expectedFeedbackParagraphs) {
```

Comment:
If i is an index, use "for (var i = 0; i < this._expectedFeedbackParagraphs.length; i++)" or use forEach. In general, use "for (var i in XXX)" only for iterating over keys in a dict.

Reply:
Fixed in upcoming commit.